apps/submission/src/controllers/submissionController.ts

```diff
@@ -393,12 +393,54 @@ const deleteEntityName = validateRequest(
 			const submissionId = Number(req.params.submissionId);
 			const user = req.user;
 			const authEnabled = authConfig.enabled;
+			const index = req.query.index;
+			const actionType = req.params.actionType.toUpperCase();
+			const entityName = req.query.entityName;
+
+			let parsedIndex;
+
+			if (parsedIndex != undefined) {
+				parsedIndex = Number(index);
+				if (isNaN(parsedIndex)) {
+					throw new lyricProvider.utils.errors.BadRequest('Index must be a valid number');
+				}
+				if (parsedIndex < 0) {
+					throw new lyricProvider.utils.errors.BadRequest('Index cannot be negative');
+				}
+			}
 
 			const submission = await lyricProvider.services.submission.getSubmissionById(submissionId);
 			if (!submission) {
 				throw new lyricProvider.utils.errors.BadRequest(`Submission '${submissionId}' not found`);
 			}
 
+			let actionObj;
+
+			if (actionType === SUBMISSION_ACTION_TYPE.enum.INSERTS) {
+				actionObj = submission.data?.inserts;
+			} else if (actionType == SUBMISSION_ACTION_TYPE.enum.UPDATES) {
+				actionObj = submission.data?.updates;
+			} else if (actionType == SUBMISSION_ACTION_TYPE.enum.DELETES) {
+				actionObj = submission.data?.deletes;
+			} else {
+				throw new lyricProvider.utils.errors.BadRequest(`Invalid actionType '${actionType}'`);
+			}
+
+			if (!actionObj) {
+				throw new lyricProvider.utils.errors.BadRequest(`Action type '${actionType}' not found in submission data`);
+			}
+			const entityObj = actionObj[entityName];
+
+			if (!entityObj) {
+				throw new lyricProvider.utils.errors.NotFound(`Entity with name '${entityName}' not found`);
+			}
+
+			const records = (entityObj as any)?.records ?? [];
```

Comment:
We don't want to type cast `any`.

Reply:
Heres a type safe solution:
```suggestion
        let records;

	// This means its of type SubmissionUpdateData[] | SubmissionDeleteData[]
	if (Array.isArray(entityObj)) {
		records = entityObj;
	} else {
		records = entityObj.records;
	}

```